instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -29,13 +29,19 @@ def process(commands) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, M
             attributes['peer.service'] = config[:peer_service] if config[:peer_service]
             attributes.merge!(OpenTelemetry::Instrumentation::Redis.attributes)
 
+
+            # Parse commands if we need to derive span attributes from them
+            if config[:db_bytes_transmitted] || config[:db_statement] != :omit
+              bytes_sent, parsed_commands = parse_commands(commands) 
+            end
+
             unless config[:db_statement] == :omit
-              parsed_commands = parse_commands(commands)
               parsed_commands = OpenTelemetry::Common::Utilities.truncate(parsed_commands, MAX_STATEMENT_LENGTH)
-              parsed_commands = OpenTelemetry::Common::Utilities.utf8_encode(parsed_commands, binary: true)
```

Comment:
Moved this  to body of `parsed_commands`

Reply:
Moving it has swapped the order of truncation and encoding. I'm not sure that's the right thing to do here. 🤔 @robertlaurin do you remember why we did it in this order, and whether it matters in practice?

Reply:
The ordering was driven by "don't waste time/resources encoding all of it if you are going to truncate it"

Reply:
We could do some benchmarks to see how much it matters in practice. Similar approach to here would probably be reasonable https://github.com/open-telemetry/opentelemetry-ruby/pull/760

Reply:
I undid the relocation, so did not feel it necessary to benchmark as part of this PR